frontend/src/components/Common/ChartFormComponents/ChartDateRangeSelector.tsx

```diff
@@ -0,0 +1,132 @@
+import { Typography, Input, makeStyles } from '@material-ui/core';
+import DatePicker from 'react-datepicker';
+import { useSafeTranslation } from 'i18n';
+
+interface ChartDateRangeSelectorProps {
+  startDate: number | null;
+  endDate: number | null;
+  onStartDateChange: (date: number | null) => void;
+  onEndDateChange: (date: number | null) => void;
+  disabled?: boolean;
+  stacked?: boolean;
+}
+
+function ChartDateRangeSelector({
+  startDate,
+  endDate,
+  onStartDateChange,
+  onEndDateChange,
+  disabled = false,
+  stacked = false,
+}: ChartDateRangeSelectorProps) {
+  const classes = useStyles();
+  const { t } = useSafeTranslation();
+
+  const handleStartDateChange = (date: Date | null) => {
+    onStartDateChange(date?.getTime() || startDate);
+  };
+
+  const handleEndDateChange = (date: Date | null) => {
+    onEndDateChange(date?.getTime() || endDate);
+  };
+
+  return (
+    <div className={classes.container}>
+      <Typography className={classes.label} variant="body2">
+        {t('Date Range')}
+      </Typography>
+      <div
+        className={classes.dateFieldsRow}
+        style={{ flexDirection: stacked ? 'column' : 'row' }}
+      >
+        <div className={classes.dateField}>
+          <Typography className={classes.fieldLabel} variant="body2">
+            {t('Start')}
+          </Typography>
+          <DatePicker
+            selected={startDate ? new Date(startDate) : null}
+            onChange={handleStartDateChange}
+            maxDate={new Date()}
```

Comment:
Not sure if this is just carried over from the existing implementation, but this does allow for invalid ranges (e.g. end date earlier than start date)